lxd/storage/backend_lxd.go

```diff
@@ -171,8 +173,75 @@ func (b *lxdBackend) CreateInstanceFromCopy(inst Instance, src Instance, snapsho
 	return ErrNotImplemented
 }
 
+// createInstanceSymlink creates a symlink in the instance directory to the instance's mount path.
+func (b *lxdBackend) createInstanceSymlink(inst Instance, mountPath string) error {
+	symlinkPath := inst.Path()
+	if !shared.PathExists(symlinkPath) {
+		err := os.Symlink(mountPath, symlinkPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// CreateInstanceFromImage creates a new volume for an instance populated with the image requested.
 func (b *lxdBackend) CreateInstanceFromImage(inst Instance, fingerprint string, op *operations.Operation) error {
```

Comment:
Hmm, so the logic you have in there is fine for dir but won't work for others.

The image unpacking logic (the filler) should be moved into a private function on lxdBackend and then have it wired in `ImageCreate` so that all the backends with OptimizedImages may work.

Then `CreateInstanceFromImage` needs to grow a few checks:
 - If we don't support OptimizedImages, it should call what you have now, using the external filler function
 - If we have OptimizedImages, then it should call driver.HasVolume to see if we've already unpacked it, if we didn't, call ImageCreate and then call CreateVolumeFromCopy to create the volume from the now existing image volume.